main/src/cgeo/geocaching/models/Geocache.java

```diff
@@ -1753,6 +1753,9 @@ public static void storeCache(final Geocache origCache, final String geocode, fi
                 }
             }
 
+            // Need to wait for images loading since HtmlImage.getDrawable is non-blocking here
```

Comment:
I believe it would be sufficient to wait before the final sendEmptyMessage(DONE). In case the handler has been disposed (the user cancelled), we can return without waiting for the images to download (instead we could try to cancel the imgGetter to before those returns).

I'm not sure if that makes much of a difference, however. If at all, probably only in batch mode.

Reply:
Good point, I was not checking cancel case. And it seems to have bigger problems underneath. Even without my change I'm getting some exception in the console

``` W/cgeo: [OkHttp] Failure (2 ms) GET https://ssl.webpack.de/gcvote.com/getVotes.php?version=cgeo&userName=lega4&password=***&cacheIds=f565d35e-2443-4ce1-b17e-0830804a55e1 (java.io.IOException: Canceled)
 W/System.err: io.reactivex.exceptions.UndeliverableException: The exception could not be delivered to the consumer because it has already canceled/disposed the flow or the exception has nowhere to go to begin with. Further reading: https://github.com/ReactiveX/RxJava/wiki/What's-different-in-2.0#error-handling | java.io.IOException: Canceled
      at io.reactivex.plugins.RxJavaPlugins.onError(RxJavaPlugins.java:367)
```
With it also similar
```
E/cgeo: [refresh--3] Geocache.storeCache
    java.lang.RuntimeException: java.lang.InterruptedException

...and a lot more lines
```

But again, seems to be a bigger issue in general. Also it looks like the way I handled exceptions in suApi is a bit wrong, but since there are no documented guidelines for this... Can we maybe have a chat about proper way to implement handling of cancellations?

As for this PR, I moved this line before `DataStore.saveCache(cache, EnumSet.of(SaveFlag.DB));` (because we need to save images before saving the cache), should be ok.